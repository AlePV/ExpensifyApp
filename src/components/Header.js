import React from "react";
import {NavLink} from "react-router-dom";
import {connect} from "react-redux";
import {startLogout} from "../actions/auth";

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
            <p> <NavLink to="/dashboard" activeClassName="isActive">Home</NavLink> </p>
            <p> <NavLink to="/create" activeClassName="isActive">Add Expense</NavLink> </p>
            <button onClick={startLogout}>Logout</button>
    </header>
);

// for the bold letters --> shows where you are
const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);

// Create private state for user data --> Private Firebase Data
// '--> Change the firebase rules
// '--> user/userUid/expenses/idOfExpense