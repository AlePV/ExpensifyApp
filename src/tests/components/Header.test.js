// Allows to render components in javascript code and assert something on what got rendered
import React from "react";
import {shallow} from "enzyme";
// import ReactShallowRenderer from "react-test-renderer/shallow";// shallow version does not include user interaction
import Header from "../../components/Header";


test("should render Header component correctly", () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();


    // expect(wrapper.find("h1").text()).toBe("Expensify");
    // const renderer = new ReactShallowRenderer;
    // // this creates a new instance of ReactShallowRenderer
    // renderer.render(<Header/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});



// returns renderedOutput of the JSX I put in 

// SNAPSHOTS --> allow to track changes to data overtime