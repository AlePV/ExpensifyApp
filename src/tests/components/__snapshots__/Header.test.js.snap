// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <p>
     
    <NavLink
      activeClassName="isActive"
      ariaCurrent="true"
      to="/dashboard"
    >
      Home
    </NavLink>
     
  </p>
  <p>
     
    <NavLink
      activeClassName="isActive"
      ariaCurrent="true"
      to="/create"
    >
      Add Expense
    </NavLink>
     
  </p>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
